ulib/experimental/Steel.Effect.Atomic.fsti

```diff
@@ -155,25 +167,27 @@ let if_then_else_ens (#a:Type) (#pre_f:pre_t) (#pre_g:pre_t) (#post_f:post_t a)
 
 let if_then_else (a:Type)
   (o:inames)
+  (#framed:eqtype_as_type bool)
   (#[@@@ framing_implicit] pre_f:pre_t) (#[@@@ framing_implicit] pre_g:pre_t)
   (#[@@@ framing_implicit] post_f:post_t a) (#[@@@ framing_implicit] post_g:post_t a)
   (#[@@@ framing_implicit] req_then:req_t pre_f) (#[@@@ framing_implicit] ens_then:ens_t pre_f a post_f)
   (#[@@@ framing_implicit] req_else:req_t pre_g) (#[@@@ framing_implicit] ens_else:ens_t pre_g a post_g)
   (#[@@@ framing_implicit] s_pre: squash (can_be_split pre_f pre_g))
   (#[@@@ framing_implicit] s_post: squash (equiv_forall post_f post_g))
-  (f:repr a o unobservable pre_f post_f req_then ens_then)
-  (g:repr a o unobservable pre_g post_g req_else ens_else)
+  (f:repr a framed o unobservable pre_f post_f req_then ens_then)
+  (g:repr a framed o unobservable pre_g post_g req_else ens_else)
```

Comment:
Can you please remind me (and also perhaps add a comment?) why do we require both to have the same framed bit? E.g. why can't we say that both may have different framed bits, and then the final computation is not framed?

Reply:
I think that if you do this, you might end up with two frames to infer for the same computation, which does not match our model or framing tactic (same problem in Steel.Effect, since the reasoning is the same).
This would happen for instance if f is already framed and you state that `if b then f else ...` is not framed. To the frame inside `f`, you would add a frame to the if_then_else when composing it with outer computations.

The better way to do this is to add additional binders/frames for f and g, and to state that the if_then_else is always framed. But this requires modifications to typecheck the if_then_else combinator w.r.t subcomp, as we discussed in the past.